pkgs/racket-doc/scribblings/reference/hashes.scrbl

```diff
@@ -986,4 +986,115 @@ one for the result.
+  ;; Filtering key-value pairs where the key is less than 3 and value is even
+  (hash-filter (for/hash ([num '(1 2 3 4 5)]) (values num (* num 2)))
+               (λ (k v) (and (< k 3) (even? v))))
+
+  ;; Filtering key-value pairs from an empty hash table
+  (hash-filter (make-hash) (λ (k v) (< k 3)))
+
+  ;; Filtering with eq? hash table based on specific key-value conditions
+  (hash-filter (make-hasheq '([#f . "false"] [#t . "true"]))
+               (λ (k v) (and (eq? k #t) (string=? v "true"))))
+
+  ;; Filtering key-value pairs where the key is a list and the value is a symbol
+  (hash-filter (hash (list 1 2) 'pair (vector 3 4) 'vector)
+               (λ (k v) (and (list? k) (symbol? v))))
+
+  ;; Filtering key-value pairs of mixed types based on custom logic
+  (hash-filter (hash "one" 1 2 "two" "three" 3)
+               (λ (k v) (and (not (number? k)) (number? v) (> v 1))))
+]
+
+@history[#:added "8.12.0.9"]
+}
+
+@defproc[(hash-filter-keys [ht hash?] [pred procedure?])
+         hash?]{
+
+Filters the @racket[hash?] @racket[ht] based on a predicate @racket[pred] applied to its keys.
+This function constructs a new hash table that includes only those key-value pairs
+from the input @racket[ht] for which the predicate @racket[pred] returns true when
+applied to the keys. Similar to @racket[hash-filter-values], the output hash table
+maintains the mutability and key comparator of the input hash table, ensuring that
+the structural and operational properties of the original hash are retained.
+
+@examples[
+  #:eval the-eval
+  ;; Filtering keys less than 3 from a hash table
+  (hash-filter-keys (for/hash ([num '(1 2 3 4 5)]) (values num 0)) (λ (k) (< k 3)))
+
+  ;; Filtering keys from an empty hash table
+  (hash-filter-keys (make-hash) (λ (k) (< k 3)))
+
+  ;; Filtering with eq? hash table
+  (hash-filter-keys (make-hasheq '([#f . "false"] [#t . "true"])) (λ (k) (eq? k #t)))
+
+  ;; Filtering lists as keys
+  (hash-filter-keys (hash (list 1 2) 'pair (vector 3 4) 'vector) list?)
+
+  ;; Filtering keys of mixed types: numbers and strings
+  (hash-filter-keys (hash "one" 1 2 "two" "three" 3) (lambda (k) (number? k)))
+
+  ;; Filtering keys that are symbols
+  (hash-filter-keys (hash 'apple "fruit" 'carrot "vegetable" "banana" "fruit")
+                    (lambda (k) (symbol? k)))
+]
+
+@history[#:added "8.12.0.9"]
+}
+
+
+@defproc[(hash-filter-values [ht hash?] [pred procedure?])
+         hash?]{
+
+Filters the @racket[hash?] @racket[ht] based on a predicate @racket[pred] applied to its values.
+This function returns a new hash table containing only the key-value pairs for which
+the predicate @racket[pred] returns true when applied to the values of @racket[ht].
+The resulting hash table retains the mutability and the key comparison predicate
+(e.g., @racket[eq?], @racket[eqv?], @racket[equal-always?], @racket[equal?]) of the input
+hash table @racket[ht]. This ensures that the characteristics of the input hash are preserved
```

Comment:
Nit (same as above): I would remove "This ensures that the characteristics of the input hash are preserved in the output, allowing for a seamless transition between different types of hash tables."